api/envoy/config/endpoint/v3/endpoint.proto

```diff
@@ -77,6 +77,12 @@ message ClusterLoadAssignment {
     //
     // Envoy supports only one element and will NACK if more than one element is present.
     // Other xDS-capable data planes will not necessarily have this limitation.
+    //
+    // In Envoy, this **drop_overloads** config can be overridden by a runtime key
+    // "config.drop_overload_support". This runtime key can be set to any integer
```

Comment:
IIUC DropOverload has a category and percentage. How does the "category" part set in the runtime-key?

Reply:
Envoy support only one category. So, no need to specify it in the key as it is basically a description text.

Reply:
I think there might be a breaking-change if in the future Envoy will support multiple categories.

Reply:
There are some discussions in supporting multiple categories in Envoy, but eventually that was put down. I am not sure Envoy will ever support multiple categories. The original API design plan to have other drop types like RLS, DOS,...,, but in Envoy those are implemented by separate filters, not through EDS.

Reply:
The other thing is that if Envoy will support multiple categories, we can have one runtime config override all categories.  Then it won't be an issue.

Reply:
> There are some discussions in supporting multiple categories in Envoy, but eventually that was put down. I am not sure Envoy will ever support multiple categories. The original API design plan to have other drop types like RLS, DOS,...,, but in Envoy those are implemented by separate filters, not through EDS.

I agree we don't know what the future holds, but I think that if anyone will need the multiple-categories feature, they will be blocked by the current runtime-key.

> The other thing is that if Envoy will support multiple categories, we can have one runtime config override all categories. Then it won't be an issue.

This actually raises another issue with this implementation.
At the moment, the EDS configuration is per-cluster. The runtime-key, on the other hand, is for all clusters in an Envoy instance. IMHO in multi-tenant environments it makes this runtime-key not very useful, as its granularity is different.

FWIW, I think one of the main differences between a runtime-feature-flag and a runtime-key is that the former is typically removed after some time, whereas the latter will stay supported for a long time. This implies that runtime-key should align with the (future) implementation, similar to API features.

Reply:
I think the runtime-key is more of a testing thing, so, granularity is not a necessity.

Reply:
Not sure I fully understand the logic behind this. The config is set by the config-server (which it can just remove when it is being sent), and if this is only needed for some testing-period then why not to use a simple runtime-feature?
The more permanent complexity that is added here, the mode challenging it will be to maintain this code.